app/services/CalculationService.scala

```diff
@@ -120,6 +123,15 @@ trait CalculationService {
       round("up", improvementsAmt))
   }
 
+  def calculateGainRebased
```

Comment:
This method is not being called. 

This can either be removed if you are going to call the calculateGainFlat method direct, or the case statement for the rebased leg should be changed to call calculateGainRebased rather than calculateGainFlat.